appstudio-controller/controllers/appstudio.redhat.com/environment_controller.go

```diff
@@ -152,8 +160,55 @@ func (r *EnvironmentReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 const (
 	SnapshotEnvironmentBindingConditionErrorOccurred = "ErrorOccurred"
 	SnapshotEnvironmentBindingReasonErrorOccurred    = "ErrorOccurred"
+	EnvironmentConditionErrorOccurred                = "ErrorOccurred"
+	EnvironmentReasonErrorOccurred                   = "ErrorOccurred"
 )
 
+// Update Status.Condition field of Environment
+func updateStatusConditionOfEnvironment(ctx context.Context, client client.Client, message string,
+	environment *appstudioshared.Environment, conditionType string,
+	status metav1.ConditionStatus, reason string) error {
+	// Check if condition with same type is already set, if Yes then check if content is same,
+	// If content is not same update LastTransitionTime
+	index := -1
+	for i, Condition := range environment.Status.Conditions {
+		if Condition.Type == conditionType {
+			index = i
+			break
+		}
+	}
+
+	now := metav1.Now()
+
+	if index == -1 {
+		environment.Status.Conditions = append(environment.Status.Conditions,
+			metav1.Condition{
+				Type:               conditionType,
+				Message:            message,
+				LastTransitionTime: now,
+				Status:             status,
+				Reason:             reason,
+			})
+	} else {
+		if environment.Status.Conditions[index].Message != message &&
```

Comment:
This `if` is redundant, since the LastTransitionTime is being set to `now` a few lines below anyway.

However, the status condition should only be updated if something has actually changed.  I think the logic should be changed such that the `else` above is an `else if`, and it should look something like
```
	} else if environment.Status.Conditions[index].Message != message ||
		environment.Status.Conditions[index].Reason != reason ||
		environment.Status.Conditions[index].Status != status {

		environment.Status.Conditions[index].Reason = reason
		environment.Status.Conditions[index].Message = message
		environment.Status.Conditions[index].Status = status
		environment.Status.Conditions[index].LastTransitionTime = now

	}
```
Note that the else if test uses `||` so that the status condition will be updated if any of the values have changed.

Also, if this else branch is executed, we should probably ensure that the `client.Status().Update()` below is only called if something has actually changed.